app/src/components/assets/AssetType.tsx

```diff
@@ -0,0 +1,152 @@
+import { IonCard, IonCardContent, IonCardHeader, IonCardSubtitle, IonCardTitle, IonCol, IonImg, IonItem, IonLabel, IonList, IonRow, IonSlide, IonSlides, useIonViewWillLeave } from "@ionic/react";
+import { useEffect, useRef, useState } from "react";
+import { useParams } from "react-router";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import Api from "../../controllers/Api";
+import { fileExtensionToIcon, formatSize, s3url } from "../../globals/functions";
+import Page from "../../pages/Page";
+import { faArrowRight, faBan, faFlag } from "@fortawesome/free-solid-svg-icons";
+import axios from "axios";
+import { AssetTypeData } from "../../globals/interfaces";
+
+const AssetType = () => {
```

Comment:
with the async data in terms of data flow, cause that currently means its not cached if you go on and off the page is it? I am not quite sure the best way data flow like this should work though, maybe @Jbithell can suggest? It may be this is correct and I am chatting out my arse

Reply:
React context

Reply:
Converted to React contexts!